server/src/main/java/org/opensearch/search/DocValueFormat.java

```diff
@@ -260,7 +265,14 @@ public String getWriteableName() {
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeString(formatter.pattern());
+            if (out.getVersion().before(Version.V_3_0_0) && formatter.equals(DateFieldMapper.getDefaultDateTimeFormatter())) {
```

Comment:
yup, thanks for pointing it out! I would like to point out that I've to go through the `getter` function route because static fields are getting initialised before the initialisation of settings module where featureflags are intialized so defining a static default formatter based on featurflag was not possible since till that time featureflags were not initialized from yaml. Also, we can't selectively use the date formatter in painless script function because script module is intialized before the setting module initialization, so I've kept the legacy formatter there

Reply:
I see, thank you